pkg/controller/workload/kubernetes/target/target.go

```diff
@@ -0,0 +1,127 @@
+limitations under the License.
+*/
+
+package target
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	util "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	runtimev1alpha1 "github.com/crossplaneio/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplaneio/crossplane-runtime/pkg/logging"
+	"github.com/crossplaneio/crossplane-runtime/pkg/meta"
+	"github.com/crossplaneio/crossplane-runtime/pkg/resource"
+	computev1alpha1 "github.com/crossplaneio/crossplane/apis/compute/v1alpha1"
+	workloadv1alpha1 "github.com/crossplaneio/crossplane/apis/workload/v1alpha1"
+)
+
+const (
+	controllerName   = "autotarget.workload.crossplane.io"
+	reconcileTimeout = 1 * time.Minute
+
+	errGetKubernetesCluster = "unable to get KubernetesCluster"
+	errCreateOrUpdateTarget = "unable to create or update KubernetesTarget"
+	errTargetConflict       = "cannot establish control of existing KubernetesTarget"
+)
+
+var log = logging.Logger.WithName("controller." + controllerName)
+
+func clusterIsBound(obj runtime.Object) bool {
+	r, ok := obj.(*computev1alpha1.KubernetesCluster)
+	if !ok {
+		return false
+	}
+
+	return r.GetBindingPhase() == runtimev1alpha1.BindingPhaseBound
+}
+
+// Controller is responsible for adding the KubernetesTarget auto-creation
+// controller and its corresponding reconciler to the manager with any runtime configuration.
+type Controller struct{}
+
+// SetupWithManager creates a new Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func (c *Controller) SetupWithManager(mgr ctrl.Manager) error {
+	r := &Reconciler{
+		kube: mgr.GetClient(),
+	}
+
+	p := resource.NewPredicates(clusterIsBound)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(controllerName).
+		For(&computev1alpha1.KubernetesCluster{}).
+		WithEventFilter(p).
+		Complete(r)
+}
+
+// A Reconciler creates KubernetesTargets for KubernetesClusters.
+type Reconciler struct {
+	kube client.Client
+}
+
+// Reconcile attempts to create a KubernetesTarget for a KubernetesCluster.
+func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error) {
+	log.V(logging.Debug).Info("reconciling", "kind", workloadv1alpha1.KubernetesTargetKindAPIVersion, "request", req)
+
+	ctx, cancel := context.WithTimeout(context.Background(), reconcileTimeout)
+	defer cancel()
+
+	cluster := &computev1alpha1.KubernetesCluster{}
+	if err := r.kube.Get(ctx, req.NamespacedName, cluster); err != nil {
+		if kerrors.IsNotFound(err) {
+			return reconcile.Result{Requeue: false}, nil
+		}
+		return reconcile.Result{Requeue: false}, errors.Wrap(err, errGetKubernetesCluster)
```

Comment:
We should requeue in case `Get` results in an error other than `NotFound`

Reply:
We actually would requeue here (the error overrides setting to `false` explicitly), but I agree that it is more in accordance with our patterns to omit `false` if we intend a requeue to always occur.